internal/service/resourcepolicy/model_test.go

```diff
@@ -0,0 +1,189 @@
+package resourcepolicy_test
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/service/resourcepolicy"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/atlas-sdk/v20240805003/admin"
+)
+
+var (
+	//go:embed testdata/policy_clusterForbidCloudProvider.json
+	clusterForbidCloudProviderJSON string
+	//go:embed testdata/policy_clusterForbidCloudProvider_no_name.json
+	clusterForbidCloudProviderNoNameJSON string
+	//go:embed testdata/policy_multipleEntries.json
+	policyMultipleEntriesJSON string
+	userIDCreate              = "65def6f00f722a1507105ad8"
+	userNameCreate            = "mvccpeou"
+	userIDUpdate              = "65def6f00f722a1507105ad9"
+	userNameUpdate            = "updateUser"
+	createdDate               = "2024-09-10T14:59:34Z"
+	lastUpdatedDate           = "2024-09-10T14:59:35Z"
+	orgID                     = "65def6ce0f722a1507105aa5"
+	policyID                  = "66e05ed6680f032312b6b22b"
+	policyBody                = "\n\tforbid (\n\t\tprincipal,\n\t\taction == cloud::Action::\"cluster.createEdit\",\n\t\tresource\n\t) when {\n\t\tcontext.cluster.cloudProviders.containsAny([cloud::cloudProvider::\"aws\"])\n\t};"
+	policies0ID               = "66e05ed6680f032312b6b22a"
+	version                   = "v1"
+)
+
+type sdkToTFModelTestCase struct {
+	SDKRespJSON     string
+	userIDCreate    string
+	userIDUpdate    string
+	userNameCreate  string
+	userNameUpdate  string
+	createdDate     string
+	lastUpdatedDate string
+	orgID           string
+	policyID        string
+	policyBody      string
+	policies0ID     string
+	version         string
+}
+
+func TestResourcePolicySDKToTFModelFull(t *testing.T) {
+	testCases := map[string]sdkToTFModelTestCase{
+		// try no name
+		"clusterForbidCloudProvider": {
+			SDKRespJSON:     clusterForbidCloudProviderJSON,
+			userIDCreate:    userIDCreate,
+			userIDUpdate:    userIDUpdate,
+			userNameCreate:  userNameCreate,
+			userNameUpdate:  userNameUpdate,
+			createdDate:     createdDate,
+			lastUpdatedDate: lastUpdatedDate,
+			orgID:           orgID,
+			policyID:        policyID,
+			policyBody:      policyBody,
+			policies0ID:     policies0ID,
+			version:         version,
```

Comment:
q: is there a reason for not having a single expected TF model result here that can be asserted as we do in other unit tests? Currently each property is in the test case and then requires a particular assertion.

Reply:
Good callout.
I started that way to create the test "iteratively" to avoid having to develop the full TFModel.
However, I agree it is cleaner to do a single equal operator.
I refactored in 574799e672009523245228daa67a73537b8609f9 and during the refactoring, I felt the need for helpers, so I created 0313baed184cf1887884d834130bd51285b987c3
WDYT?

Reply:
Nice, thank you. 
I do still feel that `createTFModel` is doing very similar logic to the function we are actual testing `NewTFResourcePolicyModel`, which is why I like the idea of making the creation of the expected TF model as trivial as possible (ideally inlined in the test case directly). There might be some complications to what I am proposing so let me know your thoughts.

Reply:
Thanks for the conversation. Led to improvements.
I choose to keep the `createTFModel` as it simplifies distinguishing the test cases.